### DISCUSSION TITLE
"Type '{ children: Element[]; }' has no properties in common with type 'IntrinsicAttributes & RefAttributes<any>'." and "Type '{ placeholder: string; value: string; onChange: (event: any) => void; className: string; }' is not assignable to type 'IntrinsicAttributes & RefAttributes<any>'.   Property 'placeholder' does not exist on type 'IntrinsicAttributes & RefAttributes<any>'." in Data Table component.

### DISCUSSION THREAD

**Discussion by matthewmngi** [shadcn-ui/ui#4704] [Q&A]
Hi, I'm using the Data Table component w/ Typescript, but I keep getting a crazy amount of errors related to "Intrinsic Attributes" & "RefAttributes". I don't know what is happening and can't find anyone else running into the same error.

Here's my code for "data-table.tsx" that it told me to write in the documentation"

```
"use client"

import * as React from "react"
import {
  ColumnDef,
  ColumnFiltersState,
  SortingState,
  flexRender,
  getCoreRowModel,
  getFilteredRowModel,
  getPaginationRowModel,
  getSortedRowModel,
  useReactTable,
} from "@tanstack/react-table"
import { Button } from "@/components/ui/button"
import { Input } from "@/components/ui/input"
import {
  Table,
  TableBody,
  TableCell,
  TableHead,
  TableHeader,
  TableRow,
} from "@/components/ui/table"



interface DataTableProps<TData, TValue> {
  columns: ColumnDef<TData, TValue>[]
  data: TData[]
}

export function DataTable<TData, TValue>({
  columns,
  data,
}: DataTableProps<TData, TValue>) {
  const [sorting, setSorting] = React.useState<SortingState>([])
  const [columnFilters, setColumnFilters] = React.useState<ColumnFiltersState>(
    []
  )

  const table = useReactTable({
    data,
    columns,
    getCoreRowModel: getCoreRowModel(),
    getPaginationRowModel: getPaginationRowModel(),
    onSortingChange: setSorting,
    getSortedRowModel: getSortedRowModel(),
    onColumnFiltersChange: setColumnFilters,
    getFilteredRowModel: getFilteredRowModel(),
    state: {
      sorting,
      columnFilters,
    },
  })

  return (
    <div>
      <div className="flex items-center py-2">
        <Input
          placeholder="Search for summary..."
          value={(table.getColumn("title")?.getFilterValue() as string) ?? ""}
          onChange={(event) =>
            table.getColumn("title")?.setFilterValue(event.target.value)
          }
          className="max-w-sm"
        />
      </div>
      <div className="rounded-md border">
        <Table>
          <TableHeader>
            {table.getHeaderGroups().map((headerGroup) => (
              <TableRow key={headerGroup.id}>
                {headerGroup.headers.map((header) => {
                  return (
                    <TableHead key={header.id}>
                      {header.isPlaceholder
                        ? null
                        : flexRender(
                            header.column.columnDef.header,
                            header.getContext()
                          )}
                    </TableHead>
                  )
                })}
              </TableRow>
            ))}
          </TableHeader>
          <TableBody>
            {table.getRowModel().rows?.length ? (
              table.getRowModel().rows.map((row) => (
                <TableRow
                  key={row.id}
                  data-state={row.getIsSelected() && "selected"}
                >
                  {row.getVisibleCells().map((cell) => (
                    <TableCell key={cell.id}>
                      {flexRender(cell.column.columnDef.cell, cell.getContext())}
                    </TableCell>
                  ))}
                </TableRow>
              ))
            ) : (
              <TableRow>
                <TableCell colSpan={columns.length} className="h-24 text-center">
                  No results.
                </TableCell>
              </TableRow>
            )}
          </TableBody>
        </Table>
      </div>
      <div className="flex items-center justify-end space-x-2 py-2">
        <Button
          variant="link"
          size="sm"
          onClick={() => table.previousPage()}
          disabled={!table.getCanPreviousPage()}
        >
          Previous
        </Button>
        <Button
          variant="link"
          size="sm"
          onClick={() => table.nextPage()}
          disabled={!table.getCanNextPage()}
        >
          Next
        </Button>
      </div>
    </div>
  )
}

```

Anyone know what's going on?

**Comment by vierynugroho** [comment]
> Hi, I'm using the Data Table component w/ Typescript, but I keep getting a crazy amount of errors related to "Intrinsic Attributes" & "RefAttributes". I don't know what is happening and can't find anyone else running into the same error.
> 
> Here's my code for "data-table.tsx" that it told me to write in the documentation"
> 
> ```
> "use client"
> 
> import * as React from "react"
> import {
>   ColumnDef,
>   ColumnFiltersState,
>   SortingState,
>   flexRender,
>   getCoreRowModel,
>   getFilteredRowModel,
>   getPaginationRowModel,
>   getSortedRowModel,
>   useReactTable,
> } from "@tanstack/react-table"
> import { Button } from "@/components/ui/button"
> import { Input } from "@/components/ui/input"
> import {
>   Table,
>   TableBody,
>   TableCell,
>   TableHead,
>   TableHeader,
>   TableRow,
> } from "@/components/ui/table"
> 
> 
> 
> interface DataTableProps<TData, TValue> {
>   columns: ColumnDef<TData, TValue>[]
>   data: TData[]
> }
> 
> export function DataTable<TData, TValue>({
>   columns,
>   data,
> }: DataTableProps<TData, TValue>) {
>   const [sorting, setSorting] = React.useState<SortingState>([])
>   const [columnFilters, setColumnFilters] = React.useState<ColumnFiltersState>(
>     []
>   )
> 
>   const table = useReactTable({
>     data,
>     columns,
>     getCoreRowModel: getCoreRowModel(),
>     getPaginationRowModel: getPaginationRowModel(),
>     onSortingChange: setSorting,
>     getSortedRowModel: getSortedRowModel(),
>     onColumnFiltersChange: setColumnFilters,
>     getFilteredRowModel: getFilteredRowModel(),
>     state: {
>       sorting,
>       columnFilters,
>     },
>   })
> 
>   return (
>     <div>
>       <div className="flex items-center py-2">
>         <Input
>           placeholder="Search for summary..."
>           value={(table.getColumn("title")?.getFilterValue() as string) ?? ""}
>           onChange={(event) =>
>             table.getColumn("title")?.setFilterValue(event.target.value)
>           }
>           className="max-w-sm"
>         />
>       </div>
>       <div className="rounded-md border">
>         <Table>
>           <TableHeader>
>             {table.getHeaderGroups().map((headerGroup) => (
>               <TableRow key={headerGroup.id}>
>                 {headerGroup.headers.map((header) => {
>                   return (
>                     <TableHead key={header.id}>
>                       {header.isPlaceholder
>                         ? null
>                         : flexRender(
>                             header.column.columnDef.header,
>                             header.getContext()
>                           )}
>                     </TableHead>
>                   )
>                 })}
>               </TableRow>
>             ))}
>           </TableHeader>
>           <TableBody>
>             {table.getRowModel().rows?.length ? (
>               table.getRowModel().rows.map((row) => (
>                 <TableRow
>                   key={row.id}
>                   data-state={row.getIsSelected() && "selected"}
>                 >
>                   {row.getVisibleCells().map((cell) => (
>                     <TableCell key={cell.id}>
>                       {flexRender(cell.column.columnDef.cell, cell.getContext())}
>                     </TableCell>
>                   ))}
>                 </TableRow>
>               ))
>             ) : (
>               <TableRow>
>                 <TableCell colSpan={columns.length} className="h-24 text-center">
>                   No results.
>                 </TableCell>
>               </TableRow>
>             )}
>           </TableBody>
>         </Table>
>       </div>
>       <div className="flex items-center justify-end space-x-2 py-2">
>         <Button
>           variant="link"
>           size="sm"
>           onClick={() => table.previousPage()}
>           disabled={!table.getCanPreviousPage()}
>         >
>           Previous
>         </Button>
>         <Button
>           variant="link"
>           size="sm"
>           onClick={() => table.nextPage()}
>           disabled={!table.getCanNextPage()}
>         >
>           Next
>         </Button>
>       </div>
>     </div>
>   )
> }
> ```
> 
> Anyone know what's going on?

What did you do to solve this problem?

**Reply by matthewmngi** [reply]
I just used the normal table component. However, I had v0.dev make a more advanced one with some of the features I wanted from the data table component.
